public/app/features/dashboard/custom_time_ranges/TimeRangeEditor.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { PureComponent } from 'react';
+
+export interface Props {
+  range: any;
+  mode: string;
+  add: any;
+  update: any;
```

Comment:
Please do proper typing on the Props

Reply:
I've created type for range but i don't know what type to put to functions (add and update)

Reply:
Think it should be something like:  `(state: { mode: string}) => void`